server/integration_tests/standalone/adversarial.py

```diff
@@ -0,0 +1,459 @@
+    with open(stats_overview_file, "w", newline="") as out:
+      out.write(str(stats))
+
+    # close writers
+    for writer in report_writers.values():
+      writer.close()
+
+  def compute_stats_from_file(
+      self,
+      results_csv_file: str,
+      stats: StatsResult,
+      reports_dir: str = None,
+      report_writers: dict[str, ResultsFileWriter] = None) -> StatsResult:
+    logging.info("Reading results csv file: %s", results_csv_file)
+    with open(results_csv_file, "r") as file:
+      for csv_row in csv.DictReader(file):
+        result = Result.from_csv_row(csv_row)
+        # logging.info("Inspecting: %s", result)
+        stats.inspect_result(result)
+        self.write_result_to_report_files(result,
+                                          reports_dir=reports_dir,
+                                          report_writers=report_writers)
+
+    return stats
+
+  def write_result_to_report_files(
+      self, result: Result, reports_dir: str,
+      report_writers: dict[str, csv.DictWriter]) -> None:
+    if not reports_dir or report_writers is None:
+      return
+    if result.status in STATUS_WITH_RESULTS:
+      self.write_result_to_report_file(result=result,
+                                       report_file_key=str(result.status),
+                                       reports_dir=reports_dir,
+                                       report_writers=report_writers)
+      if result.llm_detection_type:
+        self.write_result_to_report_file(
+            result=result,
+            report_file_key=result.llm_detection_type,
+            reports_dir=reports_dir,
+            report_writers=report_writers)
+
+  def write_result_to_report_file(self, result: Result, report_file_key,
+                                  reports_dir: str,
+                                  report_writers: dict[str, ResultsFileWriter]):
+    writer = report_writers.get(report_file_key)
+    if not writer:
+      writer = ResultsFileWriter(file_key=report_file_key, file_dir=reports_dir)
+      report_writers[report_file_key] = writer
+    writer.write(result)
+
+  def run_queries_from_files_in_dir(self, input_dir: str, output_dir: str):
+    for file_name in sorted(os.listdir(input_dir)):
+      if file_name.endswith('.tsv'):
+        input_path = os.path.join(input_dir, file_name)
+        output_path = os.path.join(output_dir, f"{file_name[0:-4]}.csv")
+        self.run_queries_from_file(input_path, output_path)
+
+  def run_queries_from_file(self, input_csv_file, output_csv_file):
+    rows = read_tsv(input_csv_file)
+    queries = [row[0] for row in rows]
+    with open(output_csv_file, "w", newline="") as out:
+      logging.info("Writing results to: %s", output_csv_file)
+      csv_writer = csv.DictWriter(out,
+                                  fieldnames=Result.get_csv_columns(),
+                                  lineterminator="\n")
+      csv_writer.writeheader()
+      with multiprocessing.Pool() as pool:
```

Comment:
what's the parallelism?

Reply:
Number of CPUs. https://screenshot.googleplex.com/4cCRAgbBy9Y8TS2